### ISSUE TITLE
"Internal Error" when signing document in v8.0.0-rc8

### ISSUE THREAD

**Issue by DazeEnd** [LibreSign/libresign#2535]
**Describe the bug**
When a logged in user attempts to sign a document (using "click to sign"), an internal error prevents the document from being signed.

**To Reproduce**
1. While logged into Nextcloud as an admin user, request a signature on a document from a regular Nextcloud user.
2. Log out of Nextcloud.
3. Log into Nextcloud as the regular user that received the signature request.
4. Open the LibreSign app, select the document from step 1, and click the "Sign" button.
5. Define your signature, if it hasn't yet been done.
6. Click the "Sign the document" button, then click the "Confirm" button.

**Expected behavior**
The document should be successfully signed.

**Actual behavior**
A message appears that reads, "Internal error. Contact admin."

**Screenshots**
![Screenshot from 2024-03-18 18-39-32](https://github.com/LibreSign/libresign/assets/2699247/60089517-2bc4-4cd9-87c5-40bd4af37442)
![image](https://github.com/LibreSign/libresign/assets/2699247/3d398328-ee77-4105-b96f-245587813958)
![image](https://github.com/LibreSign/libresign/assets/2699247/6706208d-6ee3-428f-8ef1-371d6b7b8575)

**Environment information (please complete the following information):**
 - OS: AlmaLinux 8.9
 - Browser: Firefox
 - LibreSign Version: v8.0.0-rc8
 - Nextcloud Server Version: 28.0.2
 - Logs: These messages were logged to nextcloud.log when I attempted to sign the document:
```
{"reqId":"ZfjFBLCXynYGi6UMLEUMiAAAAM4","level":3,"time":"2024-03-18T18:49:40-04:00","remoteAddr":"69.174.135.197","user":"charles@relanet.com","app":"PHP","method":"POST","url":"/ocs/v2.php/apps/libresign/api/v1/sign/uuid/3e7e15c6-6f32-490a-af99-b13f95ef89cf","message":"Undefined array key \"profileElementId\" at /var/www/newton-test.relanet.com/nextcloud/public_html/apps/libresign/lib/Helper/ValidateHelper.php#306","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0","version":"28.0.2.5","data":{"app":"PHP"}}
{"reqId":"ZfjFBLCXynYGi6UMLEUMiAAAAM4","level":3,"time":"2024-03-18T18:49:41-04:00","remoteAddr":"69.174.135.197","user":"--","app":"libresign","method":"POST","url":"/ocs/v2.php/apps/libresign/api/v1/sign/uuid/3e7e15c6-6f32-490a-af99-b13f95ef89cf","message":"Error at JSignPdf side. LibreSign can not do nothing. Follow the error message: Certificate is Empty or Invalid.","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0","version":"28.0.2.5","data":{"app":"libresign"}}
{"reqId":"ZfjFBLCXynYGi6UMLEUMiAAAAM4","level":3,"time":"2024-03-18T18:49:41-04:00","remoteAddr":"69.174.135.197","user":"--","app":"libresign","method":"POST","url":"/ocs/v2.php/apps/libresign/api/v1/sign/uuid/3e7e15c6-6f32-490a-af99-b13f95ef89cf","message":"Certificate is Empty or Invalid.","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0","version":"28.0.2.5","data":{"app":"libresign"}}
{"reqId":"ZfjFBLCXynYGi6UMLEUMiAAAAM4","level":3,"time":"2024-03-18T18:49:41-04:00","remoteAddr":"69.174.135.197","user":"--","app":"libresign","method":"POST","url":"/ocs/v2.php/apps/libresign/api/v1/sign/uuid/3e7e15c6-6f32-490a-af99-b13f95ef89cf","message":"[{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/apps\\/libresign\\/lib\\/Handler\\/JSignPdfHandler.php\",\"line\":120,\"function\":\"signWrapper\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\JSignPdfHandler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/apps\\/libresign\\/lib\\/Handler\\/JSignPdfHandler.php\",\"line\":92,\"function\":\"signUsingVisibleElements\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\JSignPdfHandler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/apps\\/libresign\\/lib\\/Handler\\/Pkcs12Handler.php\",\"line\":150,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\JSignPdfHandler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/apps\\/libresign\\/lib\\/Service\\/SignFileService.php\",\"line\":290,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Handler\\\\Pkcs12Handler\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/apps\\/libresign\\/lib\\/Controller\\/SignFileController.php\",\"line\":111,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Service\\\\SignFileService\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/apps\\/libresign\\/lib\\/Controller\\/SignFileController.php\",\"line\":80,\"function\":\"sign\",\"class\":\"OCA\\\\Libresign\\\\Controller\\\\SignFileController\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/lib\\/private\\/AppFramework\\/Http\\/Dispatcher.php\",\"line\":230,\"function\":\"signUsingUuid\",\"class\":\"OCA\\\\Libresign\\\\Controller\\\\SignFileController\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/lib\\/private\\/AppFramework\\/Http\\/Dispatcher.php\",\"line\":137,\"function\":\"executeController\",\"class\":\"OC\\\\AppFramework\\\\Http\\\\Dispatcher\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/lib\\/private\\/AppFramework\\/App.php\",\"line\":184,\"function\":\"dispatch\",\"class\":\"OC\\\\AppFramework\\\\Http\\\\Dispatcher\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/lib\\/private\\/Route\\/Router.php\",\"line\":315,\"function\":\"main\",\"class\":\"OC\\\\AppFramework\\\\App\",\"type\":\"::\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/ocs\\/v1.php\",\"line\":65,\"function\":\"match\",\"class\":\"OC\\\\Route\\\\Router\",\"type\":\"->\"},{\"file\":\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/ocs\\/v2.php\",\"line\":23,\"args\":[\"\\/var\\/www\\/newton-test.relanet.com\\/nextcloud\\/public_html\\/ocs\\/v1.php\"],\"function\":\"require_once\"}]","userAgent":"Mozilla/5.0 (X11; Linux x86_64; rv:123.0) Gecko/20100101 Firefox/123.0","version":"28.0.2.5","data":{"app":"libresign"}}
```

**Additional context**
N/A

**Comment by DazeEnd** [comment]
I just noticed that LibreSign creates a `filename.signed.pdf` file, even though the signature fails, and the document status is "available for signature".
![image](https://github.com/LibreSign/libresign/assets/2699247/dd79086e-3ebe-42de-91dd-1022d7b4d922)

**Comment by vitormattos** [comment]
<img width="1376" alt="Screenshot 2024-03-18 at 22 47 24" src="https://github.com/LibreSign/libresign/assets/36524318/f8f0c709-d6f1-460e-9d45-940ccdea959a">
<img width="1528" alt="Screenshot 2024-03-18 at 22 46 51" src="https://github.com/LibreSign/libresign/assets/36524318/d1d48aeb-c9a6-420b-abe4-51effe943a5e">

Internal server error (422 error code), when trying to sign the document


_Originally posted by @glimz in https://github.com/LibreSign/libresign/issues/2109#issuecomment-2005054930_

@glimz is this reported related with this issue? Did you made the same steps?

**Comment by vitormattos** [comment]
Making a quick look at your log I identified that:

Error at JSignPdf side. LibreSign can not do nothing. Follow the error message: Certificate is Empty or Invalid.

https://github.com/LibreSign/libresign/blob/a9395eb3b42dd2918bacc20c472df19d06070b58/lib/Handler/JSignPdfHandler.php#L140

The message`Certificate is Empty or Invalid` is because JSignPdf don't found the .pfx file. The default folder of .pfx file is `LibreSign/signature.pfx`. I will check if have a point that don't cover the scenario to get the content of signature.pfx file to generate the problem that JSignPdf receive an empty or not found certificate file.

**Comment by DazeEnd** [comment]
So do you think this is a bug, or is this a configuration problem with my installation? I can try to reinstall if you think the problem is on my end.

**Comment by vitormattos** [comment]
I need to make more tests because you aren't the unique reporting this problem. I will check what I can do to give best reports to prevent this internal error.

**Comment by vitormattos** [comment]
An error that I also need to try to reproduce that stay in your log is this:

`
Undefined array key \"profileElementId\"
`

In my development environment I already put a breakpoint at the point that this error was trigged but I don't identified a way to reproduce this error.

**Comment by DazeEnd** [comment]
Just an update: I reinstalled Nextcloud and LibreSign v8.0.0-rc8. After reinstalling, I still get the same error.

**Comment by vitormattos** [comment]
I think that this is the same that was solved by:
- https://github.com/LibreSign/libresign/pull/2565
Was a logic problem when try to sign without password certificate.

**Comment by vitormattos** [comment]
This:

- #2605

Fixes:

> An error that I also need to try to reproduce that stay in your log is this:
> 
> `Undefined array key \"profileElementId\"`
> 
> In my development environment I already put a breakpoint at the point that this error was trigged but I don't identified a way to reproduce this error.
> 
> I think that haven't relation with certificate but, since that visible elements is stored at filesystem, could have relation.

**Comment by vitormattos** [comment]
Fixed at https://github.com/LibreSign/libresign/releases/download/v8.0.0-rc9/libresign-v8.0.0-rc9.tar.gz

**Comment by DazeEnd** [comment]
This is fixed for me in v8.0.0-rc9.
